homeassistant/components/media_player/epson.py

```diff
@@ -0,0 +1,224 @@
+"""
+Support for Epson projector.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/media_player.epson/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    DOMAIN, MEDIA_PLAYER_SCHEMA, PLATFORM_SCHEMA, SUPPORT_NEXT_TRACK,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_SELECT_SOURCE, SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP,
+    MediaPlayerDevice)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_HOST, CONF_NAME, CONF_PORT, CONF_SSL, STATE_OFF,
+    STATE_ON)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['epson-projector==0.1.3']
+
+DATA_EPSON = 'epson'
+DEFAULT_NAME = 'EPSON Projector'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=80): cv.port,
+    vol.Optional(CONF_SSL, default=False): cv.boolean
+})
+
+SERVICE_ATTR_CMODE = 'epson_cmode'
+ATTR_CMODE = 'cmode'
+SUPPORT_CMODE = 33001
+
+SUPPORT_EPSON = SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE |\
+            SUPPORT_CMODE | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP | \
+            SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+        hass,
+        config,
+        async_add_devices,
+        discovery_info=None):
+    """Set up the Epson media player platform."""
+    if DATA_EPSON not in hass.data:
+        hass.data[DATA_EPSON] = []
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+
+    epson = EpsonProjector(
+        async_get_clientsession(hass, verify_ssl=False),
+        name, host,
+        config.get(CONF_PORT), config.get(CONF_SSL))
+    await epson.update()
+    if epson:
+        hass.data[DATA_EPSON].append(epson)
+        async_add_devices([epson], update_before_add=True)
+
+    async def async_service_handler(service):
+        """Handle for services."""
+        from epson_projector.const import (CMODE_LIST_SET)
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+        if entity_ids:
+            devices = [device for device in hass.data[DATA_EPSON]
+                       if device.entity_id in entity_ids]
+        else:
+            devices = hass.data[DATA_EPSON]
+        for device in devices:
+            if service.service == SERVICE_ATTR_CMODE:
+                cmode = service.data.get(SERVICE_ATTR_CMODE)
+                if cmode in CMODE_LIST_SET:
```

Comment:
This check can now be removed since the service validation schema is checking this.